test/Esprima.Tests.SourceGenerators/Snapshots/StringMatcherGeneratorTests.ScannerStringMatchingGeneration.01StringMatchers.g.verified.cs

```diff
@@ -389,33 +687,38 @@ public partial class Scanner
 
     internal static partial string? TryGetInternedKeyword(ReadOnlySpan<char> input)
     {
+        if ((uint) input.Length - 2 > 8)
+        {
+            return null;
+        }
+
         switch (input.Length)
         {
             case 2:
                 switch (input[1])
                 {
                     case 'f':
-                        return input.SequenceEqual("if".AsSpan()) ? "if" : null;
+                        return input[0] == 'i' && input[1] == 'f' ? "if" : null;
```

Comment:
@EgorBo any insight if vectorization could be used here to improve the perf? I think @lahma tried it without success, I would have hoped it's faster than manual checks like this. Or maybe just after specific string lengths?

Reply:
All help is always appreciated. Based on benchmarks indexing seems to be faster for these small strings that are usual case for parsing JS. This idea of branching for string lengths + discriminators was actually based on @EgoBo 's insights for fast string matching in some PR I can't remember anymore!

Reply:
@sebastienros @lahma it should be optimized in .NET 7.0, e.g.:
![image](https://user-images.githubusercontent.com/523221/185483493-1e277447-5f9e-4a44-b290-fede3f966ff3.png)

I don't remember the exact .NET 7.0 Preview it got merged, but the latest Preview7 definitely has it 🙂

Reply:
Wow,  that's a nice improvement. Sadly with netstandard target I think we need to keep the index hack and probably document that it can be removed later - not sure about the year 😄